TodoApp/src/TodoApp.Api/Controllers/ItemsController.cs

```diff
@@ -66,15 +70,32 @@ public async Task<IHttpActionResult> PostAsync(Item item)
 
         public async Task<IHttpActionResult> PutAsync(Guid id, Item item)
         {
-            var updatedItem = await _repository.UpdateAsync(id, item);
+            if (id == Guid.Empty)
+                return await PostAsync(item);
+
+            if (!await _itemCacher.ItemExists(id))
+                return NotFound();
+
+            if (!item.IsValidForUpdating())
+                return BadRequest();
+
+            var composedItem = await _itemCacher.GetItem(id);
+            composedItem.Text = item?.Text;
```

Comment:
You are changing the item in the controller and after that in the updater. Please move this change into updater.